From 53b10a2ccb2396cddaac112f98d26a2312365b31 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:10:34 -0400
Subject: [usb] Revert: pl2303: also use the divisor based baud rate encoding method for baud rates < 115200 with HX chips

Message-id: <1410887463-20674-563-git-send-email-dzickus@redhat.com>
Patchwork-id: 95478
O-Subject: [RHEL7 PATCH 562/591] Revert "usb: pl2303: also use the divisor based baud rate encoding method for baud rates < 115200 with HX chips"
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit e2afb1d66644a3c55e3a46ba312e302a065ecac5
Author: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date:   Fri Nov 1 09:18:38 2013 -0700

    Revert "usb: pl2303: also use the divisor based baud rate encoding method for baud rates < 115200 with HX chips"

    This reverts commit 61fa8d694b8547894b57ea0d99d0120a58f6ebf8.

    Revert all of the pl2303 changes that went into 3.12-rc1 and -rc2 as
    they cause regressions on some versions of the chip.  This will all be
    revisited for later kernel versions when we can figure out how to handle
    this in a way that does not break working devices.

    Reported-by: Mika Westerberg <mika.westerberg@linux.intel.com>
    Cc: Frank Schafer <fschaefer.oss@googlemail.com>
    Acked-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/pl2303.c b/drivers/usb/serial/pl2303.c
index fcc1326..2c23438 100644
--- a/drivers/usb/serial/pl2303.c
+++ b/drivers/usb/serial/pl2303.c
@@ -399,7 +399,7 @@ static void pl2303_encode_baudrate(struct tty_struct *tty,
   * 2) Divisor based method: encodes a divisor to a base value (12MHz*32)
   *    => supported by HX chips (and likely not by type_0/1 chips)
   */
- if (type != HX)
+ if (type != HX || baud <= 115200)
   baud = pl2303_baudrate_encode_direct(baud, type, buf);
  else
   baud = pl2303_baudrate_encode_divisor(baud, type, buf);
-- 
1.7.1